zebra-consensus/src/checkpoint.rs

```diff
@@ -96,6 +99,26 @@ pub const MAX_CHECKPOINT_HEIGHT_GAP: usize = 400;
 /// serialized size.
 pub const MAX_CHECKPOINT_BYTE_COUNT: u64 = 32 * 1024 * 1024;
 
+/// Convert a tip into its hash and matching progress.
+fn progress_from_tip(
+    checkpoint_list: &CheckpointList,
+    tip: Option<(block::Height, block::Hash)>,
+) -> (Option<block::Hash>, Progress<block::Height>) {
+    match tip {
+        Some((height, hash)) => {
+            if height >= checkpoint_list.max_height() {
+                (None, Progress::FinalCheckpoint)
+            } else {
+                metrics::gauge!("checkpoint.verified.height", height.0 as f64);
+                metrics::gauge!("checkpoint.processing.next.height", height.0 as f64);
+                (Some(hash), Progress::InitialTip(height))
+            }
+        }
+        // We start by verifying the genesis block, by itself
+        None => (None, Progress::BeforeGenesis),
```

Comment:
Ok, let's keep the gauge up to date, so we don't get confused next time we try to debug this code.

Reply:
Ah, I think I misunderstood your question. Since this function is called by both the initialization and the reset, I thought you were asking if we want to update the metrics for both cases. But now I think you meant if we want to update the metrics in each case of the `match`?

To be clear this code already looked like this, I just refactored it.

The first case should happen only when we start after the last checkpoint. In that case the checkpoint verifier didn't do anything, so I'd argue that we don't need to set the metric.

In the last case there is no height to use as a value for the metric, since it's "-1".

I'll leave this as is but let me know if you think it's better to change, and I can create a small PR to fix it.

Reply:
> The first case should happen only when we start after the last checkpoint. In that case the checkpoint verifier didn't do anything, so I'd argue that we don't need to set the metric.

It can also happen if we fill the last checkpoint using all inbound blocks, but that's very unlikely.

So I think you're right here.

> In the last case there is no height to use as a value for the metric, since it's "-1".

We typically use `NaN` here, because it makes the line disappear in Grafana.
`-1` would also be an acceptable alternative.

But this case is impossible right now, because finalized state verification can only fail after `Nu5`.